manage-data/data-store/data-streams/failure-store.md

```diff
@@ -0,0 +1,867 @@
+    "caused_by": {
+      "type": "illegal_argument_exception",
+      "reason": "For input string: \"invalid_text\""
+    },
+    "suppressed": [
+      {
+        "type": "cluster_block_exception",
+        "reason": "index [.fs-my-datastream-2025.05.01-000002] blocked by: [FORBIDDEN/5/index read-only (api)];"
+      }
+    ],
+    "failure_store": "failed" <4>
+  },
+  "status": 400 <5>
+}
+```
+
+1. The root cause of the problem was a mapping mismatch.
+2. The document could not be redirected because the failure store was not able to accept writes at this time due to an unforeseeable issue.
+3. The complete exception tree is present on the response.
+4. The response is annotated with a flag indicating the failure store would have accepted the document, but it was not able to.
+5. The response status is `400 Bad Request` due to the original mapping problem.
+
+
+### Searching failures [use-failure-store-searching]
+
+Once you have accumulated some failures, the failure store can be searched much like a regular data stream.
+
+:::{warning}
+Documents redirected to the failure store in the event of a failed ingest pipeline will be stored in their original, unprocessed form. If an ingest pipeline normally redacts sensitive information from a document, then failed documents in their original, unprocessed form may contain sensitive information.
+
+Furthermore, failed documents are likely to be structured differently than normal data in a data stream, and thus special care should be taken when making use of [document level security](../../../deploy-manage/users-roles/cluster-or-deployment-auth/controlling-access-at-document-field-level.md#document-level-security) or [field level security](../../../deploy-manage/users-roles/cluster-or-deployment-auth/controlling-access-at-document-field-level.md#field-level-security). Any security policies that expect to utilize these features for both regular documents and failure documents should account for any differences in document structure between the two document types.
+
+To limit visibility on potentially sensitive data, users require the [`read_failure_store`](../../../deploy-manage/users-roles/cluster-or-deployment-auth/elasticsearch-privileges.md#privileges-list-indices) index privilege for a data stream in order to search that data stream's failure store data.
+:::
+
+Searching a data stream's failure store can be done by making use of the existing search APIs available in {{es}}. 
+
+To indicate that the search should be performed on failure store data, use the [index component selector syntax](elasticsearch://reference/elasticsearch/rest-apis/api-conventions.md#api-component-selectors) to indicate which part of the data stream to target in the search operation. Appending the `::failures` suffix to the name of the data stream indicates that the operation should be performed against that data stream's failure store instead of its regular backing indices.
```

Comment:
```suggestion
To indicate that the search should be performed on failure store data, use the [index component selector syntax](elasticsearch://reference/elasticsearch/rest-apis/api-conventions.md) to indicate which part of the data stream to target in the search operation. Appending the `::failures` suffix to the name of the data stream indicates that the operation should be performed against that data stream's failure store instead of its regular backing indices.
```